azure-nextgen-ts-webapp-privateendpoint-vnet-injection/index.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.
+
+import * as azure_nextgen from "@pulumi/azure-nextgen";
+import * as pulumi from "@pulumi/pulumi";
+import * as random from "@pulumi/random";
+
+const config = new pulumi.Config();
+const resourceGroupNameParam = config.require("resourceGroupNameParam");
+
+const locationParam = config.get("locationParam") || "westus2";
+const resourceGroup = new azure_nextgen.resources.latest.ResourceGroup("resourceGroup", {
+    resourceGroupName: resourceGroupNameParam,
+    location: locationParam,
+});
+
+const serverFarmNameParam = config.get("serverFarmNameParam") || "appserverfarm";
+const skuFamilyParam = config.get("skuFamilyParam") || "P1v2";
+const skuNameParam = config.get("skuNameParam") || "P1v2";
+const skuSizeParam = config.get("skuSizeParam") || "P1v2";
+const skuTier = "PremiumV2";
+
+const serverfarm = new azure_nextgen.web.v20190801.AppServicePlan("serverfarm", {
+    kind: "app",
+    location: locationParam,
+    name: serverFarmNameParam,
+    resourceGroupName: resourceGroup.name,
+    sku: {
+        capacity: 1,
+        family: skuFamilyParam,
+        name: skuNameParam,
+        size: skuSizeParam,
+        tier: skuTier,
+    },
+});
+
+// To get a random suffix
+const rand = new random.RandomString("random", {
+    length: 5,
+    special: false,
+});
+
+// Setup backend app
+const site1NameParam = config.get("site1NameParam") || pulumi.interpolate `webapp1${rand.result}`;
+const site1 = new azure_nextgen.web.v20190801.WebApp("backendApp", {
+    kind: "app",
+    location: locationParam,
+    name: site1NameParam,
+    resourceGroupName: resourceGroup.name,
+    serverFarmId: serverfarm.id,
+    siteConfig: {
+        ftpsState: azure_nextgen.web.v20190801.FtpsState.AllAllowed,
+    },
+});
+
+const webappDnsName = ".azurewebsites.net";
+const hostnameBinding1 = new azure_nextgen.web.v20190801.WebAppHostNameBinding("hostNameBindingSite1", {
```

Comment:
Currently fails with this:
```
azure-nextgen:web/v20190801:WebAppHostNameBinding (hostNameBindingSite1):
    error: autorest/azure: Service returned an error. Status=404 Code="ParentResourceNotFound" Message="Can not perform requested operation on nested resource. Parent resource 'webapp1gcXga.azurewebsites.net' not found."
```

Reply:
The name should be just `name: site1.name,`, not the interpolated value.

However, I don't think we need these resources at all: these are built-in buildings that exist out-of-the-box. I believe we'll fail with "resource already exists" if we try to add them. Not sure why the ARM template has them, but they don't check the existence, so that's likely a no-op anyways.

Reply:
Cool - that is what I was seeing when I tries that earlier (resource already exists). Thanks for confirming. Updated.